File: codility/permCheck/permCheck.c
/* 
 * Date: 04 November 2016
 * Score: 100% Correctness, 100% Performance
 * Note: Same logic as python, but done to check out codility
 */

// you can write to stdout for debugging purposes, e.g.
// printf("this is a debug message\n");

int solution(int A[], int N) {
    // write your code in C99 (gcc 4.8.2)
    int arr[N];
    int i = 0;
    int value = 0;

    for ( i = 0; i < N; i++) {
        arr[i] = 1;
    }

    for ( i = 0; i < N; i++) {
        value = A[i] -1;
        if ( (value < N) && (arr[value]) != 0 ) {
            arr[value] = 0;
        }
        else {
            return 0;
        }
    }
    for ( i = 0; i < N; i++) {
        if (arr[i] != 0) {
            return 0;
        }
    }
    return 1;

}

File: codility/permCheck/permCheck.py
#
# Date: 04 November 2016
# Score: 100% Correctness, 100% Performance
# Time Complexity: O(N)
# Storage Complexity: O(N)
# Time used: 34 min
#

# you can write to stdout for debugging purposes, e.g.
# print "this is a debug message"

def solution(A):
    # write your code in Python 2.7
    sizeA = len(A)
    arr = [1] * sizeA

    i = 0
    while (i < sizeA):
        value = A[i]-1
        if ( value < sizeA) and (arr[value] != 0):
            arr[value] = 0
        else:
            return 0
        i = i + 1

    i = 0
    while ( i < sizeA):
        if (arr[i] != 0):
            return 0
        i = i + 1
    return 1
